### ISSUE TITLE
Zero rolls when casting a spell

### ISSUE THREAD

**Issue by anaximander23** [kakaroto/Beyond20#147]
I attempted to cast a spell, and got damage rolls of zero. My best guesses on why are:

* the spell was *green-flame blade* and it doesn't know what to do about the weapon damage
* the spell was learned via the Magic Initiate feat choosing a class I don't have levels in, and it's using a value of zero levels to decide how many dice to roll.

![image](https://user-images.githubusercontent.com/7376990/77447256-635af600-6de7-11ea-9c0c-019def10168f.png)

**Comment by anaximander23** [comment]
Actually, this may be a bug with Beyond; it looks like the damage doesn't actually show up on their page either:
![image](https://user-images.githubusercontent.com/7376990/77447821-0449b100-6de8-11ea-910f-a5b9d4e6022b.png)

**Comment by kakaroto** [comment]
Thanks for reporting this! The reason it's showing 0 is because it can't determine the actual damages. You see there are two "Fire Damage" lines but they show "[object Object]" instead of actual values.

It works for me here : 
![image](https://user-images.githubusercontent.com/27990/77447691-54178080-6dc6-11ea-8056-1088d51e7bc0.png)

So the question is why it doesn't work for you... Can you reliably reproduce this issue ? It should show you the roll results in D&D Beyond, do you see the same "[object Object]" issue in D&D Beyond or is it only in Discord? 
Is it possible to share the sheet for which the problem happens so I could try and reproduce the bug on my side ?

**Comment by kakaroto** [comment]
> Actually, this may be a bug with Beyond; it looks like the damage doesn't actually show up on their page either:

ah that's interesting. Beyond20 shouldn't show that "[object Object]" value though, but yes, if the damage isn't in D&D Beyond, then there's nothing Beyond20 can do.

**Comment by anaximander23** [comment]
I've reported a bug on the Beyond forums; I'll check again if/when they fix it and let you know the results.

**Comment by kakaroto** [comment]
make sure you don't have any customization on the spell or that it's a homebrewed spell... it's weird that it would be fine here but not on your side. 
Do let me know what the D&D Beyond folks say.

**Comment by anaximander23** [comment]
So, this happened because the character in question had a +0 spellcasting modifier for this spell (I hadn't finished assigning ability scores so they were all 10s, but in theory there are valid ways this could happen). Apparently, D&D Beyond has a check to just leave it blank when a value is zero, to avoid having "+0" on everything, which means in this case it displays the little flame symbol for fire damage, but nothing else. I don't know if Beyond are going to change it because they don't seem to consider it a bug, which means that Beyond20 may have to handle it.

**Comment by kakaroto** [comment]
That doesn't make any sense, it's not about having a +0.. yes, D&D Beyond doesn't show a +0, but in your case, it's not showing the fixed 3d8 value either, which is clearly a bug.
If they thought you were complaining about the damage not being "3d8 +mod", then yeah, they'd be right, it doesn't make sense for it to show "3d8 +0" so it just shows "3d8", in your case, it's showing completely empty. I think you need to point that out specifically.
Send a screenshot to be sure, and show them what it looks like on mine as well : 
![image](https://user-images.githubusercontent.com/27990/78625535-5532be00-785a-11ea-97f0-e4e29eb8232a.png)

**Comment by kakaroto** [comment]
Note, I just tried it with Magic Initiate Feat like you did, but I'm not seeing the same error on my side : 
![image](https://user-images.githubusercontent.com/27990/78625784-f883d300-785a-11ea-9560-10268a840794.png)

**Comment by anaximander23** [comment]
What level character is that on? For a 1st-level character with +0 intelligence, there's no damage rolls, just the int mod, which is when the issue becomes apparent.

**Comment by kakaroto** [comment]
Ha! It's the first level that did it. I had used a lvl 15 character. Now I created a lvl 1 character of human variant so I can give them the magic initiate and I'm getting the same inexistent damage in D&D Beyond.
I've reread the spell description and it does say that it only adds the modifier to the second target, but there is no dice roll on either ones. I thought it was a 1d8 + modifier as the base. My bad.

So then I agree, there is no bug in D&D Beyond at this time, and the bug that was in Beyond20's discord bot has been fixed, so it's all good now.

![image](https://user-images.githubusercontent.com/27990/78689110-f6a22a00-78c3-11ea-83c0-c57dfb3d5bee.png)


Thanks for the detailed reports and sorry for being so slow to grasp this issue!
